--- a/basic-script.sh	2019-06-27 17:21:17.654715078 +0200
+++ b/basic-script.sh	2019-06-27 17:21:17.656715032 +0200
@@ -7,7 +7,8 @@
 #SBATCH --nice=50
 #SBATCH -J 'gridjob'
 #SBATCH --get-user-env=10L
-#SBATCH -n 1
+#SBATCH -n 6
+#SBATCH --ntasks-per-node 2
 #SBATCH 
 
 # Overide umask of execution node (sometime values are really strange)
@@ -125,7 +126,7 @@
   nodename=`/bin/hostname -f`
   echo "nodename=$nodename" >> "$RUNTIME_JOB_DIAG"
 fi
-echo "Processors=1" >> "$RUNTIME_JOB_DIAG"
+echo "Processors=6" >> "$RUNTIME_JOB_DIAG"
 executable='/bin/true'
 # Check if executable exists
 if [ ! -f "$executable" ]; 
