--- a/basic-script.sh 2014-01-28 14:08:57.116276178 +0100
+++ b/basic-script.sh 2014-01-28 14:08:57.116276178 +0100
@@ -4,11 +4,13 @@
 #SBATCH -e /my/session/directory.comment
 #SBATCH -o /my/session/directory.comment
 
+#SBATCH -p nordugrid
 #SBATCH --nice=50
 #SBATCH -J 'gridjob'
 #SBATCH --get-user-env=10L
 #SBATCH -n 1
 #SBATCH 
+#SBATCH --mem-per-cpu=700
 
 # Overide umask of execution node (sometime values are really strange)
 umask 077
