--- a/basic-script.sh 2014-01-28 10:35:19.780083853 +0100
+++ b/basic-script.sh 2014-01-28 10:35:19.772083853 +0100
@@ -9,6 +9,7 @@
 #SBATCH --get-user-env=10L
 #SBATCH -n 1
 #SBATCH 
+#SBATCH --mem-per-cpu=13500

 # Overide umask of execution node (sometime values are really strange)
 umask 077
@@ -99,8 +99,34 @@
 RESULT=0
 
 if [ "$RESULT" = '0' ] ; then
+# RunTimeEnvironment function for INCREASEMEMORY_1:
+RTE_function_0 () {
+joboption_memory=$(( $(echo ${joboption_memory} + ${2} | tr -d \") ))
+}
+# RunTimeEnvironment function for INCREASEMEMORY_2:
+RTE_function_1 () {
+joboption_memory=$(( $(echo ${joboption_memory} + ${2} + ${2} | tr -d \") ))
+}
 # Running RTE scripts (stage 1)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}INCREASEMEMORY_1;"
+# Calling INCREASEMEMORY_1 function:
+RTE_function_0 1  "2000" "--test \"TESTING\\ TEST\""
+if [ $? -ne 0 ]; then
+    echo "Runtime INCREASEMEMORY_1 stage 1 execution failed." 1>&2
+    echo "Runtime INCREASEMEMORY_1 stage 1 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
+runtimeenvironments="${runtimeenvironments}INCREASEMEMORY_2;"
+# Calling INCREASEMEMORY_2 function:
+RTE_function_1 1  "5000"
+if [ $? -ne 0 ]; then
+    echo "Runtime INCREASEMEMORY_2 stage 1 execution failed." 1>&2
+    echo "Runtime INCREASEMEMORY_2 stage 1 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
 echo "runtimeenvironments=$runtimeenvironments" >> "$RUNTIME_JOB_DIAG"
 if [ ! "X$SLURM_NODEFILE" = 'X' ] ; then
   if [ -r "$SLURM_NODEFILE" ] ; then
@@ -159,6 +185,24 @@
 fi
 # Running RTE scripts (stage 2)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}INCREASEMEMORY_1;"
+# Calling INCREASEMEMORY_1 function:
+RTE_function_0 2  "2000" "--test \"TESTING\\ TEST\""
+if [ $? -ne 0 ]; then
+    echo "Runtime INCREASEMEMORY_1 stage 2 execution failed." 1>&2
+    echo "Runtime INCREASEMEMORY_1 stage 2 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
+runtimeenvironments="${runtimeenvironments}INCREASEMEMORY_2;"
+# Calling INCREASEMEMORY_2 function:
+RTE_function_1 2  "5000"
+if [ $? -ne 0 ]; then
+    echo "Runtime INCREASEMEMORY_2 stage 2 execution failed." 1>&2
+    echo "Runtime INCREASEMEMORY_2 stage 2 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
 if [ ! -z  "$RUNTIME_LOCAL_SCRATCH_DIR" ] ; then
   find ./ -type l -exec rm -f "{}" ";"
   chmod -R u+w "./"
