--- a/basic-script.sh	2019-07-22 17:40:18.291497489 +0200
+++ b/basic-script.sh	2019-07-22 17:40:18.293497443 +0200
@@ -56,8 +58,12 @@
 if [ -z "$JOB_ACCOUNTING" ]; then
     echo "Failed to use both cgroups and GNU time for resource usage accounting. Accounting relies on LRMS information only." 1>&2
 fi
+
 # Setting environment variables as specified by user
-export 'GRID_GLOBAL_JOBID=@TEST_JOB_ID@'
+export 'job_foo=foobar'
+export 'test="TEST"'
+export 'rte_foo="foobar"'
+export GRID_GLOBAL_JOBID='gsiftp://@TEST_HOSTNAME@:2811/jobs/rte0_env'
 
 RUNTIME_JOB_DIR=@TEST_SESSION_DIR@/@TEST_JOB_ID@
 RUNTIME_JOB_STDIN=/dev/null
@@ -135,8 +141,22 @@
 RESULT=0
 
 if [ "$RESULT" = '0' ] ; then
+# RunTimeEnvironment function for ENV:
+RTE_function_0 () {
+export joboption_env_1="test=\"TEST\""
+export joboption_env_2="rte_foo=\"foobar\""
+}
 # Running RTE scripts (stage 1)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}ENV;"
+# Calling ENV function: 
+RTE_function_0 1 
+if [ $? -ne 0 ]; then
+    echo "Runtime ENV stage 1 execution failed." 1>&2
+    echo "Runtime ENV stage 1 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
 echo "runtimeenvironments=$runtimeenvironments" >> "$RUNTIME_JOB_DIAG"
 if [ ! "X$SLURM_NODEFILE" = 'X' ] ; then
   if [ -r "$SLURM_NODEFILE" ] ; then
@@ -205,6 +225,15 @@
 fi
 # Running RTE scripts (stage 2)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}ENV;"
+# Calling ENV function: 
+RTE_function_0 2 
+if [ $? -ne 0 ]; then
+    echo "Runtime ENV stage 2 execution failed." 1>&2
+    echo "Runtime ENV stage 2 execution failed." 1>"${RUNTIME_JOB_DIAG}"
+    exit 1
+fi
+
 # Measuring used scratch space
 echo "usedscratch=$( du -sb "$RUNTIME_JOB_DIR" | sed "s/\s.*$//" )" >> "$RUNTIME_JOB_DIAG"
 # Cleaning up extra files in the local scratch
