--- a/basic-script.sh	2019-11-04 17:19:15.660939819 +0200
+++ b/basic-script.sh	2019-11-04 17:19:15.661939804 +0200
@@ -8,7 +8,7 @@
 #SBATCH --nice=50
 #SBATCH -J 'gridjob'
 #SBATCH --get-user-env=10L
-#SBATCH -n 1
+#SBATCH -n 6
 #SBATCH 
 
 # Overide umask of execution node (sometime values are really strange)
@@ -137,8 +140,35 @@
 RESULT=0
 
 if [ "$RESULT" = '0' ] ; then
+# RunTimeEnvironment function for RTE:
+RTE_function_0 () {
+export joboption_runtime_1='EXTRA_RTE'
+# TODO: Maybe set options?
+}
+# RunTimeEnvironment function for EXTRA_RTE:
+RTE_function_1 () {
+export joboption_count=6
+}
 # Running RTE scripts (stage 1)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}RTE;"
+# Calling RTE function: 
+RTE_function_0 1 
+if [ $? -ne 0 ]; then
+    echo "Runtime RTE stage 1 execution failed." 1>&2
+    echo "Runtime RTE stage 1 execution failed." 1>>"${RUNTIME_JOB_STDERR}"
+    exit 1
+fi
+
+runtimeenvironments="${runtimeenvironments}EXTRA_RTE;"
+# Calling EXTRA_RTE function: 
+RTE_function_1 1 
+if [ $? -ne 0 ]; then
+    echo "Runtime EXTRA_RTE stage 1 execution failed." 1>&2
+    echo "Runtime EXTRA_RTE stage 1 execution failed." 1>>"${RUNTIME_JOB_STDERR}"
+    exit 1
+fi
+
 echo "runtimeenvironments=$runtimeenvironments" >> "$RUNTIME_JOB_DIAG"
 if [ ! "X$SLURM_NODEFILE" = 'X' ] ; then
   if [ -r "$SLURM_NODEFILE" ] ; then
@@ -178,7 +208,7 @@
   nodename=`/bin/hostname -f`
   echo "nodename=$nodename" >> "$RUNTIME_JOB_DIAG"
 fi
-echo "Processors=1" >> "$RUNTIME_JOB_DIAG"
+echo "Processors=6" >> "$RUNTIME_JOB_DIAG"
 echo "Benchmark=HEPSPEC:1.0" >> "$RUNTIME_JOB_DIAG"
 executable='/bin/true'
 # Check if executable exists
@@ -208,6 +238,24 @@
 fi
 # Running RTE scripts (stage 2)
 runtimeenvironments=
+runtimeenvironments="${runtimeenvironments}RTE;"
+# Calling RTE function: 
+RTE_function_0 2 
+if [ $? -ne 0 ]; then
+    echo "Runtime RTE stage 2 execution failed." 1>&2
+    echo "Runtime RTE stage 2 execution failed." 1>>"${RUNTIME_JOB_STDERR}"
+    exit 1
+fi
+
+runtimeenvironments="${runtimeenvironments}EXTRA_RTE;"
+# Calling EXTRA_RTE function: 
+RTE_function_1 2 
+if [ $? -ne 0 ]; then
+    echo "Runtime EXTRA_RTE stage 2 execution failed." 1>&2
+    echo "Runtime EXTRA_RTE stage 2 execution failed." 1>>"${RUNTIME_JOB_STDERR}"
+    exit 1
+fi
+
 # Measuring used scratch space
 echo "usedscratch=$( du -sb "$RUNTIME_JOB_DIR" | sed "s/\s.*$//" )" >> "$RUNTIME_JOB_DIAG"
 # Cleaning up extra files in the local scratch
